src/local-ssh/ipc/extensionServiceServer.ts

```diff
@@ -198,3 +203,27 @@ export class ExtensionServiceServer extends Disposable {
         this.server.forceShutdown();
     }
 }
+
+async function isPortUsed(port: number): Promise<boolean> {
+    return new Promise<boolean>((resolve, _reject) => {
+        const server = netCreateServer();
+        server.once('error', () => {
+            resolve(true);
+        });
+        server.once('listening', () => {
+            server.close();
+            resolve(false);
+        });
+        server.listen(port);
+    });
+}
```

Comment:
vscode also uses this method but it has this comment ` Uses listen instead of connect. Is faster, but if there is another listener on 0.0.0.0 then this will take 127.0.0.1 from that listener.` do you think we should take it in consideration?
https://github.com/microsoft/vscode/blob/e686621e2ed038f1d21e61f29dd45bc5fdfa680a/src/vs/base/node/ports.ts#L150-L191

Reply:
> Uses listen instead of connect. Is faster, but if there is another listener on 0.0.0.0 then this will take 127.0.0.1 from that listener.

It is listen now. Not sure what this mean.

(I tested that listen and ping are very fast 😂)

Reply:
We use `127.0.0.1` because extension server listened to it @jeanp413